### ISSUE TITLE
Feature request: ability to return match span of keyword

### ISSUE THREAD

**Issue by kootenpv** [vi3k6i5/flashtext#14]

**Comment by vi3k6i5** [comment]
Most regex libraries also give the location of the matches found. Can this information also be provided by FlashText?

For example:

```>>> from flashtext import KeywordProcessor
>>> keyword_processor = KeywordProcessor()
>>> # keyword_processor.add_keyword(<unclean name>, <standardised name>)
>>> keyword_processor.add_keyword('Big Apple', 'New York')
>>> keyword_processor.add_keyword('Bay Area')
# Maybe something like
>>> keywords_found = keyword_processor.extract_keywords('I love New York and Bay Area.', spanInfo=True)
>>> keywords_found
>>> # {'New York': (8,15), 'Bay Area': (21,28)}```

**Comment by kootenpv** [comment]
Span means the start and end in my dictionary :-)

**Comment by vi3k6i5** [comment]
can you give example?
Like as per this example
```
'I love New York and Bay Area.'
{'New York': (7,14), 'Bay Area': (20,27)} # taking start as 0
```
Span means the start and end index position in the string.

**Comment by kootenpv** [comment]
Indeed... not sure how you want to integrate it, but you might want to add `.span = (7, 14)` to the returned object.

```python
>>> kp = KeywordProcessor(span=True)
>>> kp.add_keyword("a", "a")
>>> res = kp.extract_keyword("a")
>>> res[0]
"a"
>>> res[0].span
(0, 1)
```

**Comment by vi3k6i5** [comment]
To do this:
```
>>> res[0]
"a"
>>> res[0].span
(0, 1)
```
I will have to return custom object instead of String list. 
I think we can simply do this:
```[('New York', 8, 15), ('Bay Area', 21, 28)]
# This preserves the order in which keywords were found.```

what do you think?

**Comment by evanmiltenburg** [comment]
You could also use SpaCy for tokenization rather than rely on your own implementation. That will give you span for free, and make your code faster than searching for word boundaries char by char. Plus, it will make the code easier to adapt to other languages which may specify word boundaries in other ways. 

(I doubt that SpaCy will make the code slower, because it's in Cython. Just make sure you only load and use the tokenizer, and not the entire pipeline.)

**Comment by vi3k6i5** [comment]
@evanmiltenburg ok, I will try SpaCy out. Will keep you guys posted on this.

**Comment by kootenpv** [comment]
1. While I'm a huge fan of spacy, that's quite a big dependency.

2. Please don't do `spanInfo` but `span_info` (PEP8) :-)

3. I think that it should be easy to track the start and end based no the trie?

**Comment by evanmiltenburg** [comment]
I just mentioned it because the tokenization code looks like it could be improved. And since I use SpaCy for everything anyway, I don't mind it as a dependency :) But it's up to @vi3k6i5 to decide..

(I'm unsubscribing from this thread, so I won't receive any notifications.)

**Comment by vi3k6i5** [comment]
fixed in version 2.5

```
keyword_processor = KeywordProcessor()
keyword_processor.add_keyword('Big Apple', 'New York')
keyword_processor.add_keyword('Bay Area')
keywords_found = keyword_processor.extract_keywords('I love big Apple and Bay Area.', span_info=True)
keywords_found
# output [('Bay Area', 21, 29)]
```
